apis/kubedb/v1alpha2/mssql_webhook.go

```diff
@@ -0,0 +1,287 @@
+				m.Name, "spec.internalAuth.endpointCert, spec.internalAuth.endpointCert.issuerRef' is missing"))
+		} else if m.Spec.InternalAuth.EndpointCert != nil {
+			if m.Spec.InternalAuth.EndpointCert.IssuerRef == nil {
+				allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("internalAuth").Child("endpointCert").Child("issuerRef"),
+					m.Name, "spec.internalAuth.endpointCert.issuerRef' is missing"))
+			}
+			if len(m.Spec.InternalAuth.EndpointCert.Certificates) > 1 {
+				allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("internalAuth").Child("endpointCert").Child("certificates"),
+					m.Name, "spec.internalAuth.endpointCert.certificates' can have only one certificate"))
+			}
+		}
+	}
+
+	err := mssqlValidateVolumes(m.Spec.PodTemplate)
+	if err != nil {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("podTemplate").Child("spec").Child("volumes"),
+			m.Name,
+			err.Error()))
+	}
+
+	err = mssqlValidateVolumesMountPaths(m.Spec.PodTemplate)
+	if err != nil {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("podTemplate").Child("spec").Child("containers"),
+			m.Name,
+			err.Error()))
+	}
+
+	if m.Spec.StorageType == "" {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("storageType"),
+			m.Name,
+			"StorageType can not be empty"))
+	} else {
+		if m.Spec.StorageType != StorageTypeDurable && m.Spec.StorageType != StorageTypeEphemeral {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("storageType"),
+				m.Name,
+				"StorageType should be either durable or ephemeral"))
+		}
+	}
+
+	if len(allErr) == 0 {
+		return nil
+	}
+
+	return allErr
+}
+
+// reserved volume and volumes mounts for mssql
+var mssqlReservedVolumes = []string{
+	MSSQLVolumeNameData,
+	MSSQLVolumeNameInitScript,
+	// Add any additional reserved volume names here
+}
+
+var mssqlReservedVolumesMountPaths = []string{
+	MSSQLVolumeMountPathData,
+	MSSQLVolumeMountPathInitScript,
+	// Add any additional reserved volume mount paths here
+}
+
+func mssqlValidateVersion(m *MSSQL) error {
+	var mssqlVersion catalog.MSSQLVersion
+	err := DefaultClient.Get(context.TODO(), types.NamespacedName{
+		Name: m.Spec.Version,
+	}, &mssqlVersion)
+	if err != nil {
+		return errors.New("version not supported")
+	}
+
+	return nil
+}
+
+func mssqlValidateVolumes(podTemplate *ofst.PodTemplateSpec) error {
+	if podTemplate == nil {
+		return nil
+	}
+	if podTemplate.Spec.Volumes == nil {
+		return nil
+	}
+
+	for _, rv := range mssqlReservedVolumes {
+		for _, ugv := range podTemplate.Spec.Volumes {
+			if ugv.Name == rv {
+				return errors.New("Can't use a reserved volume name: " + rv)
+			}
+		}
+	}
+
+	return nil
+}
+
+func mssqlValidateVolumesMountPaths(podTemplate *ofst.PodTemplateSpec) error {
+	if podTemplate == nil {
+		return nil
+	}
+
+	if podTemplate.Spec.Containers == nil {
+		return nil
+	}
+
```

Comment:
Logical error. Assume a case where, User set invalid value in `podTemplate.spec.initContainers[].volumeMounts`. & Didn't set anything in `podTemplate.spec.containers`.